### PR TITLE
Cleanup GTest find and build

### PR DESCRIPTION
**Pull request by biddisco**
Use find package first and download/build external gtest only if none is detected.
    
    Note that setup_GTest is added to
      cmake/
      domain/cmake
      ryoanji/cmake
    to allow for the subtree directories to be built independently

NB. This has been tested locallly, but not extensively on other platforms/machines

### CODE REVIEW

**Comment by biddisco** [comment]
NB. please do not merge this until; gravity has been merged. I will rebase onto that first.

**Review by sekelle — APPROVED** [review]
lgtm

**Review by jgphpc — APPROVED** [review]
if all tests pass, this can be merged

**Pull request merged**
